### PR TITLE
JP-1549: AMI analyze algorithm updates

### PR DESCRIPTION
**Pull request by dmggh**
This is the first in the set of code updates for ami_analyze which uses the new algorithm; later commits will have the remainder of the module updates for this code version. This commit should NOT be merged. A later set of commits (for the next official build) will incorporate code for oifits output products, the new type of config file, and accept 3D input. Implementation of these 3 aspects still requires information from INS.

Closes #5149 
Addresses #5090 / [JP-1549](https://jira.stsci.edu/browse/JP-1549)

### CODE REVIEW

File: jwst/ami/ami_analyze_step.py [hunk 1]
```diff
@@ -17,7 +16,7 @@ class AmiAnalyzeStep(Step):
     reference_file_types = ['throughput']
 
     def process(self, input):
-        """
+        """^
```
Comment (hbushouse) [inline jwst/ami/ami_analyze_step.py:19]: Is the addition of the extra ^ character intentional?

**Review by hbushouse — APPROVED** [review]
Looks OK to me.

**Comment by hbushouse** [comment]
Regarding the statement to NOT merge this PR and that subsequent updates will be coming in B7.7, are any updates to the algorithms ready for inclusion in B7.6 sometime over the next couple weeks while B7.6 is undergoing integration and testing? Or will none of the AMI updates be ready in time for B7.6?

**Comment by dmggh** [comment]
The ami code in the repo that has already merged does not contain any of the updates.


This PR covers some of the modules updated only for the new algorithm (not for the oifits output or 3D input); I can update and commit the remainder of the modules updated for only the new algorithm by the beginning of next week.


The other updates needed (oifits and 3D input) require additional information from INS.

________________________________

**Comment by dmggh** [comment]
No, stray typo.


________________________________

**Comment by dmggh** [comment]
This is the final set of updates for ami_analyze for Build 7.6. Once all the changes in this PR have been reviewed and addressed, this code can be merged. These are algorithm updates only;  updates for oifits output and processing 3D inputs will come later.

**Review by hbushouse — APPROVED** [review]

**Comment by jdavies-st** [comment]
This should _not_ have been merged.  CI tests are failing.  And now `master` is failing.  Revert?

**Pull request merged**